modules/react/button/lib/Button.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+
+import {BaseButton, ButtonContainerProps} from './BaseButton';
+import {createComponent} from '@workday/canvas-kit-react/common';
+
+/**
+ * Extends all the style properties from Box to our buttons as well as props from ButtonContainerProps.
+ * We omit `ref` since all of our buttons use `createComponent` and already give access to `ref`.
+ * Use this type to extend and customize any one off buttons that you want full control over styling.
+ */
+export interface ButtonProps extends Omit<ButtonContainerProps, 'ref'> {}
+
+export const Button = createComponent('button')({
+  displayName: 'Button',
+  Component: (
+    {
+      children,
+      icon,
+      colors,
+      iconPosition = icon ? (children ? 'start' : 'only') : undefined,
```

Comment:
Does `only` do anything? In the primary button storybook example, iconPosition isn't reference, there's just no children.

Reply:
![image](https://github.com/Workday/canvas-kit/assets/7966550/8ce17f15-0229-461a-8b83-1bd766099a48)

Reply:
Yup. 

`only` refers to when a Button has only an icon with no children. So it maps to largeOnly, mediumOnly, smallOnly and extraSmallOnly

Reply:
Where does that value of the prop be used? In the icon only variant, iconPosition isn't defined

Reply:
Which shouldn't cause an issue for users

Reply:
It seems like I can still pass `only` to a button that has children and an icon and it breaks the padding. I'm just curious why we need `only`

Reply:
Ok i think i know what's going on! this line here in `Button.tsx`:
```
iconPosition = icon ? (children ? 'start' : 'only') : undefined,
```
Is doing some funky stuff. You're changing the value here, but down below you have this logic:
```
 {icon && iconPosition === 'only' && (
          <BaseButton.Icon
            size={size}
            iconPosition={iconPosition}
            icon={icon}
            shouldMirrorIcon={shouldMirrorIcon}
          />
        )}
```
But it currently works because when you deconstruct the props, you set it to undefined if there's no children which trumps the logic below in the render function. So adding `iconPosition="only"` does nothing on an icon only button